### PR TITLE
Fix corner cases in tests discovered after running on Github Actions

### PR DESCRIPTION
**Pull request by cmelchior**
This PR is an attempt to cleanup https://github.com/realm/realm-kotlin/pull/881

- It pulls out a number of smaller fixes to out test suit which resulted in errors when running tests on Windows.
- Improves the CI check to check for both Jenkins and GHA
- Increases the wait period for some tests that timed out when running on GHA

### CODE REVIEW

**Comment by cmelchior** [comment]
This PR is now ready for review

File: packages/test-base/src/jvmMain/kotlin/io/realm/kotlin/test/platform/PlatformUtils.kt [hunk 1]
```diff
@@ -40,7 +41,22 @@ actual object PlatformUtils {
     }
 
     actual fun deleteTempDir(path: String) {
-        File(path).deleteRecursively()
+        val rootPath: Path = Paths.get(path)
+        val pathsToDelete: List<Path> =
+            Files.walk(rootPath).sorted(Comparator.reverseOrder()).collect(Collectors.toList())
```
Comment (nhachicha) [inline packages/test-base/src/jvmMain/kotlin/io/realm/kotlin/test/platform/PlatformUtils.kt:46]: Is the sort needed to delete files first?

File: packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt [hunk 2]
```diff
@@ -81,12 +81,13 @@ class RealmTests {
         // Closing a Realm should also cleanup our default (internal) dispatchers.
         // The core notifier and the finalizer thread will never be closed.
         val expectedThreadCount = initialThreads.size + 1 /* core-notifier */ + if (finalizerRunning) 0 else 1
-        var counter = 5 // Wait 5 seconds for threads to settle
-        while (totalThreadCount() != expectedThreadCount && counter > 0) {
+        var counter = 10 // Wait 5 seconds for threads to settle
```
Comment (nhachicha) [inline packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt:84]:
```suggestion
        var counter = 10 // Wait 10 seconds for threads to settle
```

File: packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt [hunk 3]
```diff
@@ -81,12 +81,13 @@ class RealmTests {
...
+        while (newThreads().any { !it.isDaemon } && counter > 0) {
             delay(1000)
             counter--
         }
-        assertEquals(expectedThreadCount, totalThreadCount(), "Unexpected thread count after closing realm: ${newThreads()}")
+        val totalThreadCount = totalThreadCount()
```
Comment (nhachicha) [inline packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt:89]: For the sake of debugging, we should print the stack trace of all non-daemon thread to find where they're are blocking

**Review by nhachicha — APPROVED** [review]
Nice catches 🎣

File: packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/SystemNotificationTests.kt [hunk 4]
```diff
@@ -76,5 +76,9 @@ class SystemNotificationTests {
             writer1.write { copyToRealm(Sample()) }
             writer2.write { copyToRealm(Sample()) }
         }
+        writer1.close()
+        writer2.close()
+        liveRealmContext.close()
```
Comment (rorbech) [inline packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/SystemNotificationTests.kt:81]: Is this actually green on CI? I tried to clean this up as part of another PR but ended up splitting into this separate issue #1563

Comment (rorbech) [reply packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/SystemNotificationTests.kt:81]: Hmm. Maybe closing the writers individually first make it work.

Comment (cmelchior) [reply packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/SystemNotificationTests.kt:81]: Yes, this is working on GHA.

File: packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt (hunk 3 above)
Comment (cmelchior) [reply packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt:89]: I thought about this, but it would massively expand the output and so far all the cases we had for this, it was enough to get the name of the thread to figure out what was going on. So for now I think I would prefer to keep as as is.
